pkg/apis/pipeline/v1beta1/param_types.go

```diff
@@ -141,17 +141,40 @@ type ArrayOrString struct {
 
 // UnmarshalJSON implements the json.Unmarshaller interface.
 func (arrayOrString *ArrayOrString) UnmarshalJSON(value []byte) error {
+	// ArrayOrString is used for Results Value as well, the results can be any kind of
+	// data so we need to check if it is empty.
+	if len(value) == 0 {
+		arrayOrString.Type = ParamTypeString
+		return nil
+	}
 	switch value[0] {
 	case '[':
+		// We're tring to Unmarshal to []string, but for cases like []int or other types
+		// of nested array which we don't support yet, we should continue and Unmarshal
+		// it to String. If the Type being set doesn't match what it actually should be,
+		// it will be captured somewhere else.
```

Comment:
thank you for adding comments 🙏 

typo - s/tring/trying

Reply:
oh thank you!

Reply:
NIT: instead of saying it is captured somewhere else, please be explicit about the validation routine which will capture this

Reply:
sure!